Add more information in our select docs (#7177)
Minio-io/minio: docs/select/README.md

# Select API Quickstart Guide [![Slack](https://slack.minio.io/slack?type=svg)](https://slack.minio.io)
Traditional retrieval of objects is always as whole entities, i.e GetObject for a 5 GiB object, will always return 5 GiB of data. S3 Select API allows us to retrieve a subset of data by using simple SQL expressions. By using Select API to retrieve only the data needed by the application, drastic performance improvements can be achieved.

You can use Select API to query objects that have the following format properties:

- CSV, JSON and Parquet - Objects must be in CSV, JSON, or Parquet format.
- UTF-8 is the only encoding type Select API supports.
- GZIP or BZIP2 - CSV and JSON files can be compressed using GZIP or BZIP2. Select API supports columnar compression for Parquet using GZIP, Snappy, LZO, LZ4. Whole object compression is not supported for Parquet objects.
- Server-side encryption - Select API supports querying objects that are protected with server-side encryption.

> - CAST function if defined is honored, if CAST is not provided then type is inferred for such as decimal values etc.

Reviewer: This line does not belong in a list of object properties. We need to set it out of the list, and rephrase it like:

> Type inference and automatic conversion of values is performed based on the context when the value is un-typed (such as when reading CSV data). The behavior can be disabled by providing explicit casts using the CAST function.

Author: Can you provide a suggestion @donatello instead?

Author: Anyways I updated.

> - Type inference and automatic conversion of values is performed based on the context when the value is un-typed (such as when reading CSV data). CAST function if present overrides automatic conversion.

Reviewer: ```suggestion

Type inference and automatic conversion of values is performed based on the context when the value is un-typed (such as when reading CSV data). CAST function if present overrides automatic conversion.
```

We can set out this last point like this, as it is not really related to the list. We are really pointing out a difference from AWS S3 Select

# Select API Quickstart Guide [![Slack](https://slack.minio.io/slack?type=svg)](https://slack.minio.io)
Traditional retrieval of objects is always as whole entities, i.e GetObject for a 5 GiB object, will always return 5 GiB of data. S3 Select API allows us to retrieve a subset of data by using simple SQL expressions. By using Select API to retrieve only the data needed by the application, drastic performance improvements can be achieved.

> You can use Select API to query objects with following features:

Reviewer 2: ```suggestion
You can use the Select API to query objects with following features:
```

- CSV, JSON and Parquet - Objects must be in CSV, JSON, or Parquet format.

> - UTF-8 is the only encoding type Select API supports.

Reviewer 2: ```suggestion
- UTF-8 is the only encoding type the Select API supports.
```

> - GZIP or BZIP2 - CSV and JSON files can be compressed using GZIP or BZIP2. Select API supports columnar compression for Parquet using GZIP, Snappy, LZO, LZ4. Whole object compression is not supported for Parquet objects.

Reviewer 2: ```suggestion
- GZIP or BZIP2 - CSV and JSON files can be compressed using GZIP or BZIP2. The Select API supports columnar compression for Parquet using GZIP, Snappy, LZO, LZ4. Whole object compression is not supported for Parquet objects.
```

> - Server-side encryption - Select API supports querying objects that are protected with server-side encryption.

Reviewer 2: ```suggestion
- Server-side encryption - The Select API supports querying objects that are protected with server-side encryption.
```

> Type inference and automatic conversion of values is performed based on the context when the value is un-typed (such as when reading CSV data). CAST function if present overrides automatic conversion.

Reviewer 2: ```suggestion
Type inference and automatic conversion of values is performed based on the context when the value is un-typed (such as when reading CSV data). If present, the CAST function overrides automatic conversion.
```

> Type inference and automatic conversion of values is performed based on the context when the value is un-typed (such as when reading CSV data). CAST function if present overrides automatic conversion.

Reviewer 2: For any reference to any API, or function of an API, should we be using a format like `CAST` or `Select`, or will regular formatting will suffice?

Author: This is a select doc so it refers to only select, I don't think we need to be explicit about. Since the context is implicit.

```

## 4. Run the Program

> Upload a sample dataset to Minio using following commands.

Reviewer 2: ```suggestion
Upload a sample dataset to Minio using the following commands.
```

25786743
```

> For more detailed SELECT SQL reference refer [here](https://docs.aws.amazon.com/AmazonS3/latest/dev/s3-glacier-select-sql-reference-select.html)

Reviewer 2: ```suggestion
For a more detailed SELECT SQL reference, please see [here](https://docs.aws.amazon.com/AmazonS3/latest/dev/s3-glacier-select-sql-reference-select.html)
```

## 5. Explore Further
- [Use `mc` with Minio Server](https://docs.minio.io/docs/minio-client-quickstart-guide)
- [Use `mc sql` with Minio Server](https://docs.minio.io/docs/minio-client-complete-guide.html#sql)
- [Use `minio-go` SDK with Minio Server](https://docs.minio.io/docs/golang-client-quickstart-guide)
- [Use `aws-cli` with Minio Server](https://docs.minio.io/docs/aws-cli-with-minio)
- [Use `s3cmd` with Minio Server](https://docs.minio.io/docs/s3cmd-with-minio)
- [The Minio documentation website](https://docs.minio.io)

## 6. Implementation Status
- Full AWS S3 [SELECT SQL](https://docs.aws.amazon.com/AmazonS3/latest/dev/s3-glacier-select-sql-reference-select.html) syntax is supported.
- All [operators](https://docs.aws.amazon.com/AmazonS3/latest/dev/s3-glacier-select-sql-reference-operators.html) are supported.

> - All aggregation, conditional, type-conversion and strings functions are supported.

Reviewer 2: Is this string or strings?

Author: It's called strings .. it's a method `STRINGS`

Reviewer: If we are going to be pedantic about it it is "string functions" - see https://docs.aws.amazon.com/AmazonS3/latest/dev/s3-glacier-select-sql-reference-string.html

Author: updated the PR @eco-minio @donatello PTAL
